tests/test_metric.py

```diff
@@ -75,6 +75,37 @@ def test_metric_addition(self):
             "Incorrect End time after addition",
         )
 
+    def test_oldest_data_datetime_with_datetime(self):
+        with self.assertRaises(TypeError, msg="incorrect parameter type accepted"):
+            _ = Metric(self.raw_metrics_list[0][0], oldest_data_datetime="2d")
+
+        expected_start_time = Metric(self.raw_metrics_list[0][0]).metric_values.iloc[4, 0]
+        new_metric = Metric(
+            self.raw_metrics_list[0][0], oldest_data_datetime=expected_start_time
+        ) + Metric(self.raw_metrics_list[1][0])
+
+        self.assertEqual(
+            expected_start_time, new_metric.start_time, "Incorrect Start time after addition"
+        )
+        self.assertEqual(
+            expected_start_time,
+            new_metric.metric_values.iloc[0, 0],
+            "Incorrect Start time after addition (in df)",
+        )
+
+    def test_oldest_data_datetime_with_timedelta(self):
+        expected_start_time = Metric(self.raw_metrics_list[0][0]).metric_values.iloc[4, 0]
+        time_delta = (
+            Metric(self.raw_metrics_list[1][0]).metric_values.iloc[-1, 0]
+            - Metric(self.raw_metrics_list[0][0]).metric_values.iloc[4, 0]
```

Comment:
see https://lintlyci.github.io/Flake8Rules/rules/W503.html
but it seems this will soon be the right way to write this

Reply:
lol it took 2 years for the PR to get merged